Browser/Browser.aspx.cs

```diff
@@ -1705,32 +1718,26 @@ private void LoadFavIcon()
         /// Render all Directories and sub directories recursive
         /// </summary>
         /// <param name="folderInfo">The folder Info.</param>
+        /// <param name="readOnlyFolders"></param>
         /// <returns>
         /// TreeNode List
         /// </returns>
-        private TreeNode RenderFolder(FolderInfo folderInfo)
+        private TreeNode RenderFolder(FolderInfo folderInfo, IDictionary<int, IEnumerable<IFolderInfo>> readOnlyFolders)
         {
-            if (!FolderPermissionController.CanViewFolder(folderInfo))
```

Comment:
To avoid a breaking change or information leak, at this stage a check could be done to ensure that the requested folder is contained within the list of "readOnlyFolders" which, again here I would like to see that become "readableFolders" or otherwise.  If the user doesn't have the ability to view the requested folder it should return a NULL.  If the folder manager wasn't able to find the folder, it would be acceptable to return the treeNode, but we wouldn't want to do that if the user didn't have read permissions

Reply:
Hi @mitchelsellers , this validation is going to be redundant as `folderInfo` you would like to check is already taken from from `readableFolders` list.  No need to to check it twice. 
Check who invokes `RenderFolder` method, so you'll get the idea. 
I hope we are safe here. Node will not be created if user does not have `READ` permission to view this folder. 
For the rest of the changes I agree with you.

Reply:
Yup, you are correct, that switch eliminated that possibility.